check-submodule.py

```diff
@@ -66,23 +68,76 @@
     ],
     "tree-sitter-python": ["*.py"],
 }
+
+
+class JsonDiff:
+    def __init__(
+        self,
+        old_metrics: T.List[pathlib.Path],
+        new_metrics: T.List[pathlib.Path],
+        compare_dir: pathlib.Path,
+        max_workers: int,
+    ):
+        self.compare_dir = compare_dir
+        self.max_workers = max_workers
+
+        # Max number of filepaths in a sublist
+        n = math.ceil(len(old_metrics) / max_workers)
+
+        # Assign a certain number of filepaths to each worker
+        self.workers_filepaths = [
+            zip(old_metrics[i * n : (i + 1) * n], new_metrics[i * n : (i + 1) * n])
+            for i in range((len(old_metrics) + n - 1) // n)
```

Comment:
Nit: you can do this more easily if you use the third argument of range, something like `range(0, len(old_metrics), n)`.

>>> list(range(0, 100, 5))
[0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80, 85, 90, 95]